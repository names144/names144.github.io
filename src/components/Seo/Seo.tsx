import * as React from 'react';
import Helmet from 'react-helmet';

type Props = React.PropsWithoutRef<{
  title?: string,
}>;

const Seo: React.FC<Props> = ({ title }: Props) => (
  <Helmet
    titleTemplate="Nicholas Ames -- %s"
    defaultTitle="Nicholas Ames -- Software Engineer"
    title={title}>
    <html lang="en" />
    <meta charSet="utf-8" />
    <meta
      httpEquiv="x-ua-compatible"
      content="ie=edge" />
    <meta
      name="description"
      content="Personal site for Nicholas Ames, Software Engineer." />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no" />
  </Helmet>
);

export default Seo;
