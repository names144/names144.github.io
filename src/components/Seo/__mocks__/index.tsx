import * as React from 'react';

const Seo = (props: React.PropsWithoutRef<any>): JSX.Element => <mock-seo {...props} />;

export default Seo;
