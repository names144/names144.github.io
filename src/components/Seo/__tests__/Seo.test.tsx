import * as React from 'react';
import renderer from 'react-test-renderer';

import Seo from '../Seo';

jest.mock('react-helmet', () => (props: any) => <mock-react-helmet {...props} />);

describe('Seo', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<Seo title="Test" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
