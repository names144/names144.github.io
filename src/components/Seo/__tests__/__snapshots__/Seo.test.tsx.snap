// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seo renders correctly 1`] = `
<mock-react-helmet
  defaultTitle="Nicholas Ames -- Software Engineer"
  title="Test"
  titleTemplate="Nicholas Ames -- %s"
>
  <html
    lang="en"
  />
  <meta
    charSet="utf-8"
  />
  <meta
    content="ie=edge"
    httpEquiv="x-ua-compatible"
  />
  <meta
    content="Personal site for Nicholas Ames, Software Engineer."
    name="description"
  />
  <meta
    content="width=device-width, initial-scale=1, shrink-to-fit=no"
    name="viewport"
  />
</mock-react-helmet>
`;
